api/src/main/java/org/codice/ddf/admin/api/validation/SourceValidationUtils.java

```diff
@@ -17,21 +17,55 @@
 import static org.codice.ddf.admin.api.config.sources.SourceConfiguration.SOURCE_USER_PASSWORD;
 import static org.codice.ddf.admin.api.handler.ConfigurationMessage.createInvalidFieldMsg;
 import static org.codice.ddf.admin.api.services.CswServiceProperties.CSW_FACTORY_PIDS;
+import static org.codice.ddf.admin.api.services.CswServiceProperties.CSW_GMD_FACTORY_PID;
+import static org.codice.ddf.admin.api.services.CswServiceProperties.CSW_PROFILE_FACTORY_PID;
+import static org.codice.ddf.admin.api.services.CswServiceProperties.CSW_SPEC_FACTORY_PID;
 import static org.codice.ddf.admin.api.services.OpenSearchServiceProperties.OPENSEARCH_FACTORY_PID;
+import static org.codice.ddf.admin.api.services.WfsServiceProperties.WFS1_FACTORY_PID;
+import static org.codice.ddf.admin.api.services.WfsServiceProperties.WFS2_FACTORY_PID;
 import static org.codice.ddf.admin.api.services.WfsServiceProperties.WFS_FACTORY_PIDS;
 import static org.codice.ddf.admin.api.validation.ValidationUtils.validateString;
 
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
+import org.apache.commons.lang.StringUtils;
 import org.codice.ddf.admin.api.config.sources.SourceConfiguration;
+import org.codice.ddf.admin.api.configurator.Configurator;
 import org.codice.ddf.admin.api.handler.ConfigurationMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.collect.ImmutableList;
 
 public class SourceValidationUtils {
 
-    public static List<ConfigurationMessage> validateWfsFactoryPid(String factoryPid,
-            String configId) {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SourceValidationUtils.class);
+
+    private static final List<String> FACTORY_PIDS = ImmutableList.of(CSW_GMD_FACTORY_PID,
```

Comment:
Instead of hardcoding these here, with the `SourceValidationUtils` knowing about the source types it validates, I think I would rather see a registration method where the different sources pass in their factory pids to this class to tell it about them.

In fact, now that you're no longer using this statically but instantiating a new instance in each source configuration, I don't think you need knowledge of this full list at all. Rather, construct your new validation utils object with just the factory pid(s) your configuration cares about.